### DISCUSSION TITLE
Where to put test files?

### DISCUSSION THREAD

**Discussion by nedsalk** [FuelLabs/fuels-ts#1290] [Ideas]
to-be-written

**Comment by arboleya** [comment]
This conversation has been held in the past.

Tests are put side-by-side with their source counterpart

Test artifacts, when required, are put into a `/test` directory side-by-side with `/src`

> _This pattern should now be expressed across the whole code base._

**Reply by nedsalk** [reply]
> Tests are put side-by-side with their source counterpart

That's not in line with what was done in #1079 and what @danielbate mentioned in #1043:

> - Package test directories should contain appropriate feature tests

I've looked through some issues/PRs and didn't find any substantial _written_ discussions or records of internal discussions on this specific topic besides the above. Please point me to any that you know of. I've created this discussion as a place where we can summarize everything about it, voice our opinions, and then come to a conclusion that #1043 can reference and that we can reference in our PRs.

**Reply by camsjams** [reply]
General JavaScript convention is side-by-side, but whatever project may exist for test file improvements, Daniel is the one handling this afaik
